tests/utils.py

```diff
@@ -45,6 +50,8 @@ def server(handler):
             server_thread = thread_class(handler, daemon=True)
             server_thread.start()
             try:
+                # await test server connectivity before yielding a result
+                socket.create_connection(HOST, timeout=0.5).close()
```

Comment:
Use a context manager instead (at least it's clearer IMO)

Reply:
I've pushed a change with that, although am not 100% sure whether it's an improvement in this case (in most cases I'd agree that using a context manager for socket/file/other closable resources makes sense).

Reply:
Yeah after looking at it.. both have their disadvantages, like the first version is a bit tricky in the sense that you see "create" then "close" afterwards but the alternative has a `pass` ... which is not better actually.

I'll just revise my opinion and let you keep whichever version you feel is better for you.

Reply:
Thanks, yep, neither is perfect.  I'll revert it to gain some more continuous integration confidence.. but might revert it back again (or think of other ideas... ?).

Reply:
One thing that worries me about the `pass` pattern is that someone might later think "hey, this is a bug in the test - the `yield` should be inside the context-manager" -- because that's a typical pattern -- and if they make that change, tests would continue to succeed _most_ of the time (as they do before this branch), until the test flakiness reappeared again in future.

I can't think how to guard against that other than comments though, really.

Reply:
No problem.  I'm sorry for all the pings when you've been busy.  I think this is ready now, and I'll remove the 'do not merge flag' in a few moments.